carapace-server/src/main/java/org/carapaceproxy/server/mapper/StandardEndpointMapper.java

```diff
@@ -96,20 +99,39 @@ public void configure(ConfigurationStore properties) throws ConfigurationNotVali
         this.forceBackendParameter = properties.getProperty("mapper.forcebackend.parameter", forceBackendParameter);
         LOG.info("configured mapper.forcebackend.parameter=" + forceBackendParameter);
 
+        for (int i = 0; i < MAX_IDS; i++) {
+            String prefix = "header." + i + ".";
+            String id = properties.getProperty(prefix + "id", "");
+            String name = properties.getProperty(prefix + "name", "");
+            if (!id.isEmpty() && !name.isEmpty()) {
+                String value = properties.getProperty(prefix + "value", "");
+                addHeader(id, name, value);
+                LOG.info("configured header " + id + " name:" + name + ", value:" + value);
+            }
+        }
+
         for (int i = 0; i < MAX_IDS; i++) {
             String prefix = "action." + i + ".";
             String id = properties.getProperty(prefix + "id", "");
-            if (!id.isEmpty()) {
-                boolean enabled = Boolean.parseBoolean(properties.getProperty(prefix + "enabled", "false"));
+            boolean enabled = Boolean.parseBoolean(properties.getProperty(prefix + "enabled", "false"));
+            if (!id.isEmpty() && enabled) {
                 String action = properties.getProperty(prefix + "type", "proxy");
                 String file = properties.getProperty(prefix + "file", "");
                 String director = properties.getProperty(prefix + "director", DirectorConfiguration.DEFAULT);
                 int code = Integer.parseInt(properties.getProperty(prefix + "code", "-1"));
-                LOG.info("configured action " + id + " type=" + action + " enabled:" + enabled);
-                if (enabled) {
-                    ActionConfiguration config = new ActionConfiguration(id, action, director, file, code);
-                    addAction(config);
+                ActionConfiguration config = new ActionConfiguration(id, action, director, file, code);
+                String _headers = properties.getProperty(prefix + "headers", "");
+                if (!_headers.isEmpty()) {
+                    String[] _headersids = _headers.split(",");
+                    for (String headerId : _headersids) {
+                        if (!this.headers.containsKey(headerId)) {
```

Comment:
how do we deal with spaces ?
I see two options:
- "trim" the values
- "report an error" as you are currently doing